### PR TITLE
Improve parseability of type and constant string representations

### PR DESCRIPTION
**Pull request by paulsc96**
## Overview
Enable `TypePrinter` and various `toString()`s to produce more parseable output by optionally emitting well-formed SystemVerilog expressions in more situations.

All changes below except 3. keep existing stringifications unchanged by default. All existing code remains compatible through optional fields and parameters.

These changes do not guarantee type or constant parseability, but cover some simple-to-fix gaps in Slang's already quite parseable stringifications.

### `TypePrinter`
1. Add the `skipScopedTypeNames` option to `TypePrinter` to omit suffixed type names from container types.
2. Add the `fullEnumType`option to `TypePrinter` to include enum base types.

### `ConstantValue` and `SVInt`
3. Use assignment patterns (`'{...}` ) instead of simple brackets (`[...]`) for container constants.
4. Expose `abbreviateThresholdBits` to the end user of `ConstantValue.toString()` and callees to control or suppress integer vector abbreviation as needed.
5. Add the optional `exactUnknowns` parameter to `ConstantValue.toString()` and callees to ensure a reversible representation of integer vectors with unknown bits.

### CODE REVIEW

File: source/numeric/ConstantValue.cpp [hunk 1]
```diff
@@ -37,51 +37,54 @@ std::string ConstantValue::toString() const {
                 return "$"s;
             else if constexpr (std::is_same_v<T, Elements>) {
                 FormatBuffer buffer;
-                buffer.append("[");
+                buffer.append("'{");
                 for (auto& element : arg) {
-                    buffer.append(element.toString());
+                    buffer.append(element.toString(abbreviateThresholdBits));
```
Comment (MikePopoloski) [inline source/numeric/ConstantValue.cpp:42]: Did you want to pass the exactUnknowns flag down into these other toString() calls as well?

File: include/slang/ast/types/TypePrinter.h [hunk 2]
```diff
@@ -25,6 +25,8 @@ struct SLANG_EXPORT TypePrintingOptions {
     bool addSingleQuotes = false;
     bool elideScopeNames = false;
     bool printAKA = false;
+    bool skipScopedTypeNames = false;
```
Comment (MikePopoloski) [inline include/slang/ast/types/TypePrinter.h:28]: Can you make sure these are exposed in the Python bindings as well? Can be easily added in the bindings/python/types.cpp file.

File: source/numeric/SVInt.cpp [hunk 3]
```diff
@@ -637,25 +637,27 @@ std::ostream& operator<<(std::ostream& os, const SVInt& rhs) {
     return os;
 }
 
-std::string SVInt::toString() const {
+std::string SVInt::toString(bitwidth_t abbreviateThresholdBits, bool exactUnknowns) const {
     // guess the base to use
     LiteralBase base;
-    if (bitWidth < 8 || (unknownFlag && bitWidth <= 64))
+    // unless the int is wholly or digit/hexit-wise X or Z (TODO: worth checking?),
```
Comment (MikePopoloski) [inline source/numeric/SVInt.cpp:643]: Probably not worth the effort. I'd just remove the TODO.

**Review by MikePopoloski — COMMENTED** [review]
Thanks for the diff. I have a few minor comments but happy to land this once you address them.

**Comment by MikePopoloski** [comment]
Also as you can probably tell from the failed tests, there are a bunch of eval tests that need to be updated for your ConstantValue toString change to use `'{}` but I think it's fine to not make that an option and just update the tests instead.

File: source/numeric/SVInt.cpp (hunk 3 above)
Comment (paulsc96) [reply source/numeric/SVInt.cpp]: Done

File: include/slang/ast/types/TypePrinter.h (hunk 2 above)
Comment (paulsc96) [reply include/slang/ast/types/TypePrinter.h:28]: Done

File: source/numeric/ConstantValue.cpp (hunk 1 above)
Comment (paulsc96) [reply source/numeric/ConstantValue.cpp]: Yes, I did. Thanks for catching that; I propagated it.

**Comment by paulsc96** [comment]
I chose to make 3. optional now and keep the existing default behavior, maintaining full compatibility.

**Pull request merged**